src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/sdpa_fuse_transpose_reshape.cpp

```diff
@@ -0,0 +1,200 @@
+        auto k_reshape_order = as_type_ptr<op::v0::Constant>(k_reshape->get_input_node_shared_ptr(1));
+        auto v_reshape_order = as_type_ptr<op::v0::Constant>(v_reshape->get_input_node_shared_ptr(1));
+        if (k_reshape_order && v_reshape_order) {
+            if (k_reshape_order->cast_vector<int32_t>() != v_reshape_order->cast_vector<int32_t>()) {
+                return false;
+            }
+        } else if (k_reshape->get_input_node_shared_ptr(1) != v_reshape->get_input_node_shared_ptr(1)) {
+            return false;
+        }
+
+        std::shared_ptr<op::v1::Transpose> qkv_transpose[3] = {};
+        std::shared_ptr<op::v0::Constant> qkv_transpose_order[3] = {};
+        qkv_transpose[0] = as_type_ptr<op::v1::Transpose>(pattern_map.at(q_transpose_node).get_node_shared_ptr());
+        qkv_transpose[1] = as_type_ptr<op::v1::Transpose>(pattern_map.at(k_transpose_node).get_node_shared_ptr());
+        qkv_transpose[2] = as_type_ptr<op::v1::Transpose>(pattern_map.at(v_transpose_node).get_node_shared_ptr());
+        qkv_transpose_order[0] = as_type_ptr<op::v0::Constant>(pattern_map.at(q_transpose_order_node).get_node_shared_ptr());
+        qkv_transpose_order[1] = as_type_ptr<op::v0::Constant>(pattern_map.at(k_transpose_order_node).get_node_shared_ptr());
+        qkv_transpose_order[2] = as_type_ptr<op::v0::Constant>(pattern_map.at(v_transpose_order_node).get_node_shared_ptr());
+        auto out_tranpose = as_type_ptr<op::v1::Transpose>(pattern_map.at(out_transpose_node).get_node_shared_ptr());
+        auto out_transpose_order = as_type_ptr<op::v0::Constant>(pattern_map.at(out_transpose_order_node).get_node_shared_ptr());
+
+        if (!(is_expected_transpose(qkv_transpose[0]) && is_expected_transpose(qkv_transpose[1]) &&
+              is_expected_transpose(qkv_transpose[2]))) {
+            return false;
+        }
+        if (!is_expected_transpose(out_tranpose)) {
+            return false;
+        }
+
+        auto out_reshape = as_type_ptr<op::v1::Reshape>(pattern_map.at(out_reshape_node).get_node_shared_ptr());
+        if (!is_expected_reshape(out_reshape, true)) {
+            return false;
+        }
+
+        OutputVector args = {q_reshape->get_input_node_shared_ptr(0),
+                             k_reshape->get_input_node_shared_ptr(0),
+                             v_reshape->get_input_node_shared_ptr(0)};
+
+        // Config
+        intel_cpu::SDPAWithTransposeReshape::Config config;
+        config.is_causal = sdpa->get_causal();
+        config.fuse_concat = false;
+
+        const auto& permute_q = qkv_transpose_order[0]->cast_vector<int32_t>();
+        const auto& permute_k = qkv_transpose_order[1]->cast_vector<int32_t>();
+        const auto& permute_v = qkv_transpose_order[2]->cast_vector<int32_t>();
+        const auto& permute_output = out_transpose_order->cast_vector<int32_t>();
+        if (permute_q != permute_k || permute_q != permute_v || permute_q != permute_output) {
```

Comment:
Is it a redundant check since it is already checked in `is_expected_transpose` ?

Reply:
Yes, removed.